### PR TITLE
Support CLIPTokenizerFast for CLIPProcessor

### LINKED ISSUE #15888: CLIPProcessor with CLIPTokenizerFast
# 🚀 Feature request

Current `CLIPProcessor` doesn't support `CLIPTokenizerFast` requiring `CLIPTokenizer`.
In my thinking, there is no reason not to support `CLIPTokenizerFast` for `CLIPProcessor`

## Your contribution

https://github.com/huggingface/transformers/blob/v4.16.2/src/transformers/models/clip/processing_clip.py#L23

it may be easy by modifying upper python code. I think I can contribute.

### PR DESCRIPTION
**Pull request by cosmoquester**
# What does this PR do?

Fixes #15888 

Support CLIPTokenizerFast for CLIPProcessor.
Update CLIPProcessor test code for saving and loading CLIPProcessor with fast tokenizer.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/clip/test_processor_clip.py [hunk 1]
```diff
@@ -86,19 +89,28 @@ def prepare_image_inputs(self):
         return image_inputs
 
     def test_save_load_pretrained_default(self):
-        tokenizer = self.get_tokenizer()
+        tokenizer_s = self.get_tokenizer()
+        tokenizer_r = self.get_rust_tokenizer()
```
Comment (patil-suraj) [inline tests/clip/test_processor_clip.py:93]: (nit) Maybe call this `tokenizer_slow` and `tokenizer_fast`

**Review by patil-suraj — APPROVED** [review]
LGTM! Thanks a lot for working on this!

File: tests/clip/test_processor_clip.py (hunk 1 above)
Comment (cosmoquester) [reply tests/clip/test_processor_clip.py:93]: actually I use the name refering to `test_tokenization_clip.py`
like https://github.com/huggingface/transformers/blob/a7df656f0343046736a71d4b2218769db92977eb/tests/clip/test_tokenization_clip.py#L82-L83

but I agree tokenizer_slow and tokenizer_fast is better.

Comment (cosmoquester) [reply tests/clip/test_processor_clip.py]: I modified it!

**Pull request merged**